### ISSUE TITLE
Protection enabled

### ISSUE THREAD

**Issue by italobusi** [tsaad-dev/te#210]
Within the protection container it looks like there is some duplication between the enable and the protection-type: is it possible to have enable=true and protection-type="te-types:lsp-protection-unprotected"?
 
```
    container protection {
      description
        "Protection parameters.";
      leaf enable {
        type boolean;
        default "false";
        description
          "A flag to specify if LSP protection is enabled.";
        reference
          "RFC4427";
      }
      leaf protection-type {
        type identityref {
          base te-types:lsp-protection-type;
        }
        default "te-types:lsp-protection-unprotected";
        description
          "LSP protection type.";
      }
``` 

If not, the protection can be enabled by setting the protection-type<>"te-types:lsp-protection-unprotected" and the enable leaf could be removed from the model.
 
Alternatively, to keep consistency between the protection and restoration configuration, it is also possible to either:

a) Keep the enable leaf; change/remove the default value for the protection-type leaf and deprecate the te-types:lsp-protection-unprotected (already defined in RFC8776); or

b) Remove the enable leaf also from the restoration container and add a new “te-types:lsp-restoration-type” (e.g., te-types:lsp-restoration-restore-none) to disable restoration
 
See comment 9) in https://mailarchive.ietf.org/arch/msg/teas/26vHESr_cYIsUes6Fuvw3YAOtjk/

**Comment by italobusi** [comment]
**2022-10-07 TE Call**

Agreed for option b).

Note – The te-types needs to be updated accordingly

**Comment by italobusi** [comment]
- [ ] @tsaad-dev : remove the enabled leaf from the protection container in TE tunnel model
- [ ] @italobusi : add a new identity to te-types:lsp-restoration-type” (e.g., te-types:lsp-restoration-restore-none) to disable restoration

**Comment by tsaad-dev** [comment]
Addressed in 9717721

**Comment by tsaad-dev** [comment]
Add default for restoration-type to  'te-types:lsp-restoration-restore-none'

**Comment by italobusi** [comment]
> * @italobusi : add a new identity to te-types:lsp-restoration-type” (e.g., te-types:lsp-restoration-restore-none) to disable restoration

Done in PR #205 

@tsaad-dev : 

> Add default for restoration-type to 'te-types:lsp-restoration-restore-none'

**Comment by tsaad-dev** [comment]
Addressed in 902c1ad
